### PR TITLE
fix docs for custom types

### PR DESCRIPTION
**Pull request by wolverdude**
Prompted by [Issue 151](https://github.com/Pylons/colander/issues/151), this improves the accuracy and readability of the Extending Colander section of the docs.

I did not create descriptions for the methods `flatten`, `unflatten`, `set_value`, or `get_value`. That might be fodder for another PR, though I'm no longer actively working with Colander, so someone else may need to take up the cause.

### CODE REVIEW

**Comment by mmerickel** [comment]
LGTM

File: docs/extending.rst [hunk 1]
```diff
@@ -71,59 +75,78 @@ The above schema has a member named ``interested`` which will now be
...
-receive a :attr:`colander.null` value.  :attr:`colander.null` will be sent to
-the type during serialization and deserialization in circumstances where a
-value has not been provided by the data structure being serialized or
-deserialized.  In the common case, when the ``serialize`` or ``deserialize``
-method of type receives the :attr:`colander.null` value, it should just
-return :attr:`colander.null` to its caller.
+Both the ``serialize`` and ``deserialize`` methods must be able to
+receive :attr:`colander.null` values and handle them intelligently. This
+will happen whenever the data structure being serialized or deserialized
+does not provide a value for this node. In most cases, ``serialize`` or
```
Comment (tseaver) [inline docs/extending.rst]: "In the common case" is deliberately weaker than "in most cases":  perhaps "in many cases" instead?

File: docs/extending.rst [hunk 2]
```diff
@@ -137,22 +160,6 @@ within its ``deserialize`` method.
...
-schemas do, so how types are constructed is beyond the scope of
-Colander itself.
-
-The :exc:`colander.Invalid` exception may be raised during
-serialization or deserialization as necessary for whatever reason the
-type feels appropriate (the inability to serialize or deserialize a
-value being the most common case).
-
-For a more formal definition of a the interface of a type, see
-:class:`colander.interfaces.Type`.
```
Comment (tseaver) [inline docs/extending.rst]: I don't like dropping this section, as it makes clear that construction is not relevant to *`colander`*, per se.

Comment (wolverdude) [inline docs/extending.rst]: Nothing from this section has been dropped, only *moved*. See lines 26, 89, 104, and 133.

File: docs/extending.rst (hunk 1 above)
Comment (wolverdude) [inline docs/extending.rst]: That's fair. I'll change it.

File: docs/extending.rst (hunk 2 above)
Comment (tseaver) [inline docs/extending.rst]: OK, I see that the information is still present.

**Pull request merged**